src/services/sqlObject/core/dependecy.ts

```diff
@@ -1,15 +1,90 @@
 import { ISqlObject } from '../sqlObject';
 
-
 export class Dependecy {
 
     private constructor() {
     }
 
-    public static async fromObject(sqlObject: ISqlObject[]) {
+    public static async fromObjects(sqlObjects: ISqlObject[]) {
+        await sqlObjects.forEach(sqlObject =>{
+            this.getDependecys(sqlObject.definition as string, sqlObject, sqlObjects);
+        });
+        return;
+    }
+
+    private static async getDependecys(sqlObjDefinition : string, sqlObject: ISqlObject, sqlObjects: ISqlObject[]) {
+        if(sqlObjDefinition == undefined) return;
+        const extTbAndVst:Promise<string[]> = this.extractTbAndVst(sqlObjDefinition);
+        const extStpAndFnc:Promise<string[]> = this.extractStpAndFnc(sqlObjDefinition);
+        const dependecys:string[] = (await extStpAndFnc).concat(await extTbAndVst);        
+        this.areValidDependecys(dependecys, sqlObject, sqlObjects);
+        return;
+    }
+
+    private static async extractTbAndVst(sqlObjDefinition: string):Promise<string[]>{
```

Comment:
In this method, check also for `UPDATE`, `INSERT INTO`, and `DELETE`.
To give you some examples, check `StpCliOrdDet_KyUpd`, `StpCliOrdDet_KyIns` and `StpCliOrdDet_KyDel`.
If an entity has more than one dependency with the same name, add only one dependancy with that name.